pkg/deployer/manifest/ensure.go

```diff
@@ -93,12 +95,64 @@ func (m *Manifest) Reconcile(ctx context.Context) error {
 		return err
 	}
 
+	m.DeployItem.Status.ProviderStatus = statusData
+	if err := m.kubeClient.Status().Update(ctx, m.DeployItem); err != nil {
+		m.DeployItem.Status.LastError = lsv1alpha1helper.UpdatedError(m.DeployItem.Status.LastError,
+			currOp, "UpdateStatus", err.Error())
+		return err
+	}
+
+	if m.ProviderConfiguration.HealthChecks.Disable {
+		m.DeployItem.Status.Phase = lsv1alpha1.ExecutionPhaseSucceeded
+		m.DeployItem.Status.ObservedGeneration = m.DeployItem.Generation
+		m.DeployItem.Status.LastError = nil
+		return nil
+	}
+
+	return m.CheckResourcesHealth(ctx)
+}
+
+// CheckResourcesHealth checks if the managed resources are Ready/Healthy.
+func (m *Manifest) CheckResourcesHealth(ctx context.Context) error {
+	var (
+		currOp          = "CheckResourcesHealthManifests"
+		manifestDecoder = serializer.NewCodecFactory(ManifestScheme).UniversalDecoder()
+		status          = &manifest.ProviderStatus{}
+	)
+
+	if _, _, err := manifestDecoder.Decode(m.DeployItem.Status.ProviderStatus.Raw, nil, status); err != nil {
+		return err
+	}
+
+	if len(status.ManagedResources) == 0 {
+		return nil
+	}
+
+	objects := make([]*unstructured.Unstructured, len(status.ManagedResources))
+	for i, mr := range status.ManagedResources {
+		// do not check ignored resources.
+		if mr.Policy == manifest.IgnorePolicy {
+			continue
+		}
+		ref := mr.Resource
+		obj := kutil.ObjectFromTypedObjectReference(&ref)
+		objects[i] = obj
+	}
+
+	timeOut := time.Duration(m.ProviderConfiguration.HealthChecks.TimeOutSeconds) * time.Second
+	if err := health.WaitObjectsHealthy(ctx, timeOut, m.log, m.kubeClient, objects); err != nil {
+		m.DeployItem.Status.LastError = lsv1alpha1helper.UpdatedError(m.DeployItem.Status.LastError,
```

Comment:
can you adjust it to the new internal landscaper error structure.
e.g. https://github.com/gardener/landscaper/pull/119/files#diff-bfa610c54952e4b39144ec1a4ddccedfd7f8798dc3c4dc0e9a7ea32177b09e1cR61

Reply:
I've adjusted everywhere in the helm and manifest deployer